ISSUE-41: How to modify allowed file extensions during ingest

mitchellkeaney opened the pull request:
# What is this
This PR is addressing issue #41. It is a guide on how to modify a webform in order to allow more file extensions to be uploaded during ingest.

This is a new guide from scratch (new screenshots, links, voice etc.) It will definitely need some extra eyes 👀and extra brain 🧠 to be fully fleshed out.
Thanks!

mitchellkeaney committed:
first commit: adding md file and all the screenshots for the document

"How to modify allowed file extensions during ingest." Reference to ISSUE-41 in Esmero.

mitchellkeaney committed:
first solo revision before making pr against esmero

grammar and tense edits

DiegoPino commented:
@mitchellkeaney thanks. Will review in detal during the day. I already gave the docs a good first reading pass and i like the tone of your text. 

I do feel somethings are just too explicit. E.g the location of every menu (like 'manage' + screenshot' then structure + screen shot, etc) distracts a little bit and assuming we will have a lot of these guides it will become too much. 

So my first recommendation would be to think modularly, we discussed this in the past, but will higlight again but also explain why: We really don't want to repeat **Drupal 8 documentation** as our own. Reasons are, things change and we have no control of that, depending on the theme location of things look different, Drupal corrects steps, exposes modules but also means every time we release we have to change every screenshot, not something 2 people can handle by themselves and it will become a bottle neck.

## So steps/tasks here (i shared some of this in #38 )

1.- Make common screenshots re usable and reuse them. The login, logout, etc, no need to have that same screenshot copied in many different folders. We really **don't need to explain** people how to log in in every guide. If you feel we need, then you can have a  `.md` for that if you want and point them to that.
2.- Remove Version header and bottom line (black) on some of the screenshots, so we can reuse them if beta3 does not bring significant changes, e.g
3.- When speaking about settings/configs.Provide screenshots that group a few options in one? E.g the settings for adding the file extensions is a quite close up zoom, but then you speak about a setting (size) that is not visible. Let's try to have things we speak about in the screenshot present. I know that is not an exact rule and it will conflict with 1. sometimes. If it conflicts with 1, means we can discuss and make a decision

Regarding the Text: I would probably just add a little more context info. There is no mention (and a great oportunity) of how the saved file ends visible in the JSON (and how archipelago descripes/routes the files based on their type into different structures) That can be of course its own guide and can be a second pull. 
But! There is a good change that you will have to explain in this `.md` / the one you are committing, why this form element is shown and not the video on (the access conditionals) and why uploading a file does not mean you can see it. And how/what is needed (like the next step) for that. The way i see this is: always think in circles, we need to make sure people have the full path drawn upfront and somehow predict what is the next thing they will need. 

Questions? Thanks! 👍

mitchellkeaney commented:
@DiegoPino awesome! thanks for your first pass feedback and insight.

a lot of the things i had questions or ideas about you already answered or reinforced. For example, being too explicit on the direction of things (such as the "log in section" and some of the screenshots) and then overall making things more modular. i don't have any other questions at the moment that i can think of... will make a comment here if i need clarification on something.

i will make some revisions based on those 3 points you made. thanks!

DiegoPino commented:
@mitchellkeaney awesome. Good luck

mitchellkeaney committed:
revisions: new screen shots, less explicit and more contextual text

mitchellkeaney commented:
@DiegoPino made some revisions but i have a feeling the language may still be a bit verbose. let me know what you think!

mitchellkeaney committed:
grammar and language revisions

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,71 +1,60 @@
-Now, **this step is imperative for saving your changes**, scroll to the bottom of your elements list page and click `Save elements`. So again, first we hit `Save` at the bottom of the `Edit Upload Associated Documents element` page (where we added which file extensions are allowed to be uploaded) and then scroll to the bottom of the Webform elements page and click `Save elements`.
+**This next step is imperative for saving your changes**, scroll to the bottom of your elements list page and click `Save elements` in order to persist all changes made.
 
 ![Save Elements](../imgs/modifyingfileextensionsinwebform/09_save-elements.jpg)
```
Consistency over everything. There is a 07 file and then a 09 file. Patterns, patterns, patterns

DiegoPino replied:
Consistency over everything. There is a 07 file and then a 09 file. Patterns, patterns, patterns

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,71 +1,60 @@
+# Customizing Webforms: Modifying allowable file extensions
 
-This guide will walk users through how to modify their webform to allow additional file extensions to be ingested into Archipelago. For this guide, we will be adding `csv` and `md` files to be included in our workflow when creating, for example, a `DigitalDocument` object.
+A guide to walk users through how to modify the Webform `Descriptive Metadata` to allow additional file extensions to be ingested into Archipelago.
```
Ok, this actually reads much better! Make sure to link (using either the absolute path http://localhost bla bla ) or the relative in plain english (like , `which can be found at ... bla bla and is shipped by default on archipelago-deployment'). That applies to anything we are providing and that is not obvious core Drupal!

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,71 +1,60 @@
 ### Managing Webforms
 
-Let's click on `Manage`, then `Structure` and when the page loads, scroll down and you'll find `Webforms`. Go ahead and click `Webforms`.
+Once logged in as an admin, the first thing we need to do is navigate to the Wbeforms page so we can edit the Webform `Descriptive Metadata.` Click on `Manage`, then `Structure` and when the page loads, scroll down and click `Webforms`.
```
`admin` or `Admin user` but using the single quotes way so it grays out

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,71 +1,60 @@
 
-- user: admin
-- pass: archipelago
+![Default Step 4](../imgs/modifyingfileextensionsinwebform/01_default-step-4.jpg)
```
Why is this called step 4?

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,71 +1,60 @@
 ![Manage > Structure > Webforms](../imgs/modifyingfileextensionsinwebform/03_manage-structure-webforms.jpg)
 
-Here is where all of the Webforms inside your Archipelago live. In this screenshot, you'll see that I've begun creating a few custom webforms (`Digital Object Collection Webform`, `My Favorite Webform` and `Student Webform`) to suit some of my repository needs. However, let's focus on the default wenform shipped with Archipelago: `Descriptive Metadata`. This will be the webform we're going to mod ...
+This is where all of the Webforms inside your Archipelago live. In this screenshot you'll notice a few custom webforms that have been created to fit the needs of this particular repository: `Digital Object Collection Webform`, `My Favorite Webform` and `Student Webform`. For this guide we're going to edit the default Webform `Descriptive Metadata`. Go ahead and click `Build` under the `OPERATIONS ...
```
They won't have those webforms in their system. This is confusing

DiegoPino replied:
They won't have those webforms in their system. This is confusing

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,71 +1,60 @@
-**Two things to note**
-- All file extensions are separated by a space; no `,` or `.` between the values (i.e. file extensions).
-- Notice that above the `Allowed file extensions` is a `Maximum file size` field. This is where you can change the minimum or maximum file size allowed for ingest.
+**Please note:** All file extensions are separated by a space; no `,` or `.` between the values (i.e. file extensions).
```
Good

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,71 +1,60 @@
+### Recap + "What if I want to...?"
 
-When logged in as an admin, we went to *Manage > Structure > Webforms* and click `Build` under the OPERATIONS column for `Descriptive Metadata` to edit the elements in the Webform (shortcut: /admin/structure/webform/manage/descriptive_metadata). Then we click on `Upload Associated Documents` to edit the element, scroll down to the *Allowed file extensions* field and add `csv` and `md` without `.` ...
+When logged in as an admin, we go to *Manage > Structure > Webforms* and click on `Build` under the `OPERATIONS` column of `Descriptive Metadata` (shortcut: /admin/structure/webform/manage/descriptive_metadata). Then we click on `Upload Associated Documents` to edit the element, scroll down to the *Allowed file extensions* field and add `csv` without `.` or `,` separating the values. Click `Save` ...
```
Some notes: i feel this piece of documentation still lacks a little bit of generalization. Like, maybe speaking about the 'element type' which is a a file upload element. We can talk about this

DiegoPino reviewed (changes requested):
Some changes

mitchellkeaney committed:
commit 22 April

- consistent file naming p a t t e r n s
- revised the step names
- revised lanauge, added more context at the end of the md to assist in more general help / assistance outside of the guide's specific use case.

mitchellkeaney committed:
editing grammatical errors

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -1,12 +1,12 @@
+When creating an Archipelago Digital Object (ADO), there is a step during the ingest to upload the files associated with your ADO. You may notice there is a section on the Webform outlining the maximum number of files allowed, the maximum file size allowed, and the allowed file extensions that can be uploaded.
 
-![Default Step 4](../imgs/modifyingfileextensionsinwebform/01_default-step-4.jpg)
+Let's say we are creating an ADO with the media type `DigitalDocument` and this ADO contains a data set saved as a `csv` file. Archipelago has no restrictions on what file extensions can be uploaded, however some use cases will require a little configuring to  fit a specific need. This guide will walk users through the steps to modify the default Webform shipped with Archipelago, `Descriptive Met ...
```
double spacing between to and fit

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -16,48 +16,64 @@ When creating, for example, a `DigitalDocument` Object in Archipelago and you re
 ### Managing Webforms
 
-Once logged in as an admin, the first thing we need to do is navigate to the Wbeforms page so we can edit the Webform `Descriptive Metadata.` Click on `Manage`, then `Structure` and when the page loads, scroll down and click `Webforms`.
+Once logged in as `admin`, the first thing we need to do is navigate to the Wbeforms page so we can edit the Webform `Descriptive Metadata.` Click on `Manage`, then `Structure` and when the page loads, scroll down and click `Webforms`.
```
Typo: `Wbeform`

DiegoPino commented on docs/modifyingfileextensionsinwebform.md:
```diff
@@ -16,48 +16,64 @@ When creating, for example, a `DigitalDocument` Object in Archipelago and you re
+![Upload Associated Documents element](../imgs/modifyingfileextensionsinwebform/04_upload-associated-documents.jpg)
 
-A new screen will pop up named `Edit Upload Associated Documents element`. This is where you configure can configure maximum number of values (under `ELEMENT SETTINGS`), the maximum file size and edit the allowed file extensions for this element, which is what we'll be doing. The latter both exist under `FILE SETTINGS`, high lighted in the screenshot below.
+A new screen will pop up named `Edit Upload Associated Documents element`. This is where you configure can configure the maximum number of values (under `ELEMENT SETTINGS`), the maximum file size and also edit the allowed file extensions for this element, which is what we'll be doing. The latter both exist under `FILE SETTINGS`, high lighted in the screenshot below.
```
`This is where you configure can configure ` is double configure, looks like a typo

DiegoPino reviewed (changes requested):
Some typos, rest looks legit

DiegoPino commented:
@mitchellkeaney i marked some typos and syntax errors i found. The rest looks good. Please correct that so i can merge and we can move forward. Thanks a lot

mitchellkeaney committed:
correct syntax and grammatical errors

- double space between "to" and "fit"; line 9
- double "configure"; line 33
- "Webform"; line 19

mitchellkeaney commented:
thanks @DiegoPino just made another pass, fixed those errors and made a new commit. let me know if you notice anything else.

DiegoPino reviewed (approved):
Well, longest 79 lines review since we went into quarantine but finally ready to merge! I won't close ISSUE-41 until you have added a link to this guide to the table of content of the main README.md, but this pull is ready to roll. Merging! Thanks
